src/bpf/bpfd-manager.c

```diff
@@ -0,0 +1,489 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <sys/wait.h>
+
+#include "sd-daemon.h"
+#include "sd-messages.h"
+
+#include "bpf-dlopen.h"
+#include "bpfd-conf.h"
+#include "bpfd-manager.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "fs-util.h"
+#include "hashmap.h"
+#include "missing_syscall.h"
+#include "mkdir.h"
+#include "process-util.h"
+#include "signal-util.h"
+#include "socket-util.h"
+#include "stdio-util.h"
+#include "umask-util.h"
+
+static int get_memlock_by_fd(int fd, char **ret_memlock) {
+        _cleanup_free_ char *fdinfo = NULL;
+        char *memlock, *memlock_line;
+        char path[STRLEN("/proc/self/fdinfo/") + DECIMAL_STR_MAX(int)];
+        int r;
+
+        xsprintf(path, "/proc/self/fdinfo/%i", fd);
+        r = read_full_virtual_file(path, &fdinfo, NULL);
+        if (r < 0)
+                return r;
```

Comment:
None of this code is library code so please complain loudly about every error to make this easier to debug.